### PR TITLE
Fix for '+' being reserved in SQL, test fixes

### PR DESCRIPTION
**Pull request by wpride**
cc @benrudolph

### CODE REVIEW

File: modern/org/commcare/modern/database/TableBuilder.java [hunk 1]
```diff
@@ -180,7 +180,7 @@ public String getTableCreateString() {
 
     //sqlite doesn't like dashes
     public static String scrubName(String input) {
-        return input.replace("-", "_");
+        return input.replace("-", "_dash_").replace(".", "_dot_").replace("+", "_plus_");
```
Comment (phillipm) [inline modern/org/commcare/modern/database/TableBuilder.java]: Pretty sure this will require a bunch of data migrations. For example, this is used to get db column names in https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/models/database/SqlStorage.java#L152

Comment (phillipm) [inline modern/org/commcare/modern/database/TableBuilder.java]: Oh, I guess if you have verified that no current scrubbed columns etc use `-`, `.`, or `+`, then we should be good.

File: tests/test/org/commcare/util/test/SqlStorageIndexedTests.java [hunk 2]
```diff
@@ -150,13 +150,13 @@ public void testSqlLedgerStorage() {
             ledgerStorage.write(l2);
             ledgerStorage.write(l3);
 
-            Vector ids = ledgerStorage.getIDsForValue("entity_id", "ledger_entity_id");
+            Vector ids = ledgerStorage.getIDsForValue("entity-id", "ledger_entity_id");
```
Comment (phillipm) [inline tests/test/org/commcare/util/test/SqlStorageIndexedTests.java]: Can you explain why this change needed?

File: modern/org/commcare/modern/database/TableBuilder.java (hunk 1 above)
Comment (benrudolph) [inline modern/org/commcare/modern/database/TableBuilder.java]: I would've expected android to crash if it did have usernames like that. though it seems plausible that it would have come across `.` in usernames. Do you know if Android's sqlite version is just able to handle that? Or is there some other escaping process

Comment (benrudolph) [inline modern/org/commcare/modern/database/TableBuilder.java]: Ah nvm, you're saying it's used for column names (not username) I would expect that to fail

Comment (phillipm) [inline modern/org/commcare/modern/database/TableBuilder.java]: Yeah, and it would only crash if you try to update CommCare with an app installed from `2.30` to the dev build. Otherwise the tables would just get created with the new naming structure.
